storm-core/pom.xml

```diff
@@ -602,6 +612,42 @@
                           <pattern>org.eclipse.jetty</pattern>
                           <shadedPattern>org.apache.storm.shade.org.eclipse.jetty</shadedPattern>
                         </relocation>
+                        <relocation>
+                            <pattern>com.codahale.metrics</pattern>
+                            <shadedPattern>org.apache.storm.shade.com.codahale.metrics</shadedPattern>
+                        </relocation>
```

Comment:
If we are shading the codahale API the normal way of installing a custom metrics reporter is not going to work.  We should probably look at either not shading this, or at providing instructions on how to make other Reporters work.  But that can probably be in a follow on JIRA.